tests/infer/test_valid_models.py

```diff
@@ -852,79 +858,123 @@ def guide():
     assert_ok(model, guide, Trace_ELBO())
 
 
-@pytest.mark.parametrize('enumerate_', [None, "sequential", "parallel"])
-def test_enum_discrete_iarange_shape_broadcasting_ok(enumerate_):
+@pytest.mark.parametrize('enumerate_,expand', [
+    (None, True),
+    ("sequential", True),
+    ("sequential", False),
+    ("parallel", True),
+    ("parallel", False),
+])
+def test_enum_discrete_iarange_shape_broadcasting_ok(enumerate_, expand):
 
     @poutine.broadcast
-    @config_enumerate(default=enumerate_)
     def model():
         x_iarange = pyro.iarange("x_iarange", 10, 5, dim=-1)
         y_iarange = pyro.iarange("y_iarange", 11, 6, dim=-2)
         with pyro.iarange("num_particles", 50, dim=-3):
             with x_iarange:
                 b = pyro.sample("b", dist.Beta(torch.tensor(1.1), torch.tensor(1.1)))
-                assert b.shape == torch.Size((50, 1, 5))
             with y_iarange:
                 c = pyro.sample("c", dist.Bernoulli(0.5))
-                if enumerate_ == "parallel":
-                    assert c.shape == torch.Size((2, 50, 6, 1))
-                else:
-                    assert c.shape == torch.Size((50, 6, 1))
             with x_iarange, y_iarange:
                 d = pyro.sample("d", dist.Bernoulli(b))
-                if enumerate_ == "parallel":
-                    assert d.shape == torch.Size((2, 1, 50, 6, 5))
-                else:
-                    assert d.shape == torch.Size((50, 6, 5))
 
-    assert_ok(model, model, TraceEnum_ELBO(max_iarange_nesting=3,
-                                           strict_enumeration_warning=(enumerate_ == "parallel")))
+        # check shapes
+        assert b.shape == (50, 1, 5)
+        if enumerate_ == "parallel":
+            if expand:
+                assert c.shape == (2, 50, 6, 1)  # FIXME actual shape == (2, 1, 1, 1)
```

Comment:
The results seem fine to me, could you expand on the `FIXME`s here and below?

Reply:
opps, I fixed these in 53d4b30. will remove.